### DISCUSSION TITLE
Build, Deploy, and Test with VS Code Extension!

### DISCUSSION THREAD

**Discussion by emilyJLin95** [ConsenSys-archive/truffle#5148] [Announcements]
Hey Trufflers!
We're excited to announce the release of the Truffle for VS Code Extension. Microsoft VS Code users can now build, debug and deploy smart contracts on the Ethereum blockchain using the [Truffle Suite](https://twitter.com/trufflesuite) of Web3 dev tools.
Check it out [here](https://trufflesuite.com/blog/build-on-web3-with-truffle-vs-code-extension/?utm_source=github&utm_medium=devcommunity&utm_campaign=2022_May_truffle-vscode-ext-blog_announcement_content)

**Comment by dewedeth** [comment]
works like a charm

**Reply by mrghubba** [reply]
I'm having a lot of issues with Truffle for VS Code Extension.. It does not update Node JS on VSC when I try to run Truffle. I've installed everything as instructed. My system is Ubuntu 20.04 ARM64. As shown in the screenshot of Visual Studio Code, Node JS is still shown as uninstalled. Even though my terminal displays that I am using version 18.2.0 of Node JS.

Here's the Visual Studio Code Error:

https://tinyurl.com/VSCerror 

My screenshot shows evidence of what I've installed.

https://tinyurl.com/TruffleInstallation 

Please help. Thanks

**Reply by apdan510** [reply]
@mrghubba thanks for reaching out! Computer details for visibility:

Node: 14
npm: 6
CPU: ARM64
OS: ubuntu 20
Truffle npm package: installed
node tools: installed

**Reply by mrghubba** [reply]
Yes that's correct. I have Node JS version 18 installed as well. So I'm not sure why I'm getting installation errors on Visual Studio Code

**Reply by dewedeth** [reply]
> Yes that's correct. I have Node JS version 18 installed as well. So I'm not sure why I'm getting installation errors on Visual Studio Code

you need version 14

**Reply by mrghubba** [reply]
The ^ in the following pic https://paste.pics/HD9Q9 means that version 14 is the minimum. 14 or higher should work.
But I have version 14 anyway. NPM version 6 and Node JS is shown here:
https://paste.pics/HD9PB
But the Node JS button is still there on VSC.

**Reply by apdan510** [reply]
@mrghubba sorry about the delay. Waiting on my Arm64 test computer shipment for testing its been delayed ETA 3 days. :/ 
Kind regards,
Alex | Truffle Support | Consensys

**Reply by kevinbluer** [reply]
Hey @mrghubba, just to weigh on this, at present the extension only supports Node `v14` up to `v17` so if you're able to come down to a version in that range you should be all good. Also just for reference we should have support for `v18` coming relatively soon and will update here when it's available.

**Reply by mrghubba** [reply]
Hi Kevin. I have tried ALL versions.  Screenshot for NPM version 6 and Node JS version 14 is shown here:
https://paste.pics/HD9PB

**Reply by kevinbluer** [reply]
Ah weird (and apologies if I missed that point earlier). Can you mind try the following when you have a moment? When opening VSCode, select the Output tab along the bottom panel and select "Truffle for VSCode" from the dropdown. You should see output similar to the following, which if you could share that would be useful to help us troubleshoot:
  
<img width="1107" alt="Screen Shot 2022-06-30 at 5 41 53 PM" src="https://user-images.githubusercontent.com/210755/176790470-81573b35-a946-431c-93e7-3963ae14d89b.png">

**Reply by mrghubba** [reply]
Kevin, here's the Output screenshot as requested: https://paste.pics/HG74M

**Reply by kevinbluer** [reply]
Thanks for sharing @mrghubba! Weirdly it's still showing Node v18. It's possible that VScode isn't picking up the `nvm use`'ed version (slightly dated, but [this Stackoverflow entry highlights the issue](https://stackoverflow.com/questions/44700432/visual-studio-code-to-use-node-version-specified-by-nvm)). Want to try `nvm alias default 14` and confirm the version in the integrated terminal within VSCode?

<img width="142" alt="Screen Shot 2022-07-01 at 2 01 13 PM" src="https://user-images.githubusercontent.com/210755/176955136-c62b1c01-405c-4848-a453-d2935d0a85f0.png">

**Reply by kevinbluer** [reply]
Also [as mentioned](https://stackoverflow.com/a/44707192/990332), you'll likely need to restart VSCode after running `nvm alias` for the changes to be picked up 👍

**Reply by mrghubba** [reply]
Okay I tried it. But now there are 3 items showing as uninstalled when I used 'nvm alias': https://paste.pics/HG9TE  
Output Part 1 is here: https://paste.pics/HG9UG
Output Part 2 is here: https://paste.pics/HG9U8

**Reply by kevinbluer** [reply]
Ah if you could switch to Node 16 (via NVM) that will overcome the NPM version dependency issue. After that you will have to reinstall Ganache and Truffle again, as they'll it will be a different global modules directory when you switch Node version.

Beyond that fingers crossed it should be all good 🤞

**Reply by mrghubba** [reply]
Okay but please advise on the step by step process of it all. How do I uninstall Truffle and Ganache? Then please advise on the Ubuntu commands to reinstall Truffle and Ganache. I just don't want to risk any more errors.

**Reply by apdan510** [reply]
@mrghubba I can help you with this tomorrow. I have my arm64 environment set up now, but I will be back to assist 7/4/22 

Going to write out all the exact steps to remove truffle and ganache and reinstall it with node 14 and trouble shoot vscode with you. 

Kind regards,
Alex | Truffle Support

**Reply by mrghubba** [reply]
Thanks Alex (and Kevin), Much appreciated. I'll wait until tomorrow. The only thing is that, as per Kevin, we need to try Node 16 (via NVM) rather than Node 14.

**Reply by apdan510** [reply]
@mrghubba lets open a new terminal

**1. Lets set Node to 16 so that your npm will be 7.18 or higher with this command:**

**Install node 16 with nvm if it is not installed already:**
`nvm install 16`

**Set the default version:** 
`nvm alias default 16`

**Restart VS code**

**2. Uninstall truffle and ganache and then reinstall them:**
`npm uninstall -g truffle`
then
`npm uninstall -g ganache`

**Reinstall globally with:**
**npm install -g truffle --force**
then
**npm install -g ganache --force**

**3. Open VScode and try to install the truffle extension again.** 

Hope this helps!

**Reply by mrghubba** [reply]
It works! Woohoo! Thanks so much guys :-)

One question: How do I install VScode updates? I click on the update to download them. But what is the next step on an Ubuntu ARM64?

**Reply by apdan510** [reply]
@mrghubba You can update vscode by clicking the gear/settings icon on the left side dashboard then click **download update** and it should automatically install. I'm glad it works! @kevinbluer thank you again!

**Reply by kevinbluer** [reply]
Awesome news @mrghubba, thanks for sticking at it while we got things resolved! Big thanks to @apdan510 too 👌 

Also any other feedback as you start using it is very much welcome!

**Reply by mrghubba** [reply]
Thanks Alex and Kevin! I'll be sure to give some feedback on this software as time goes on. Kevin, if I have occasional technical questions, is there a way that I can contact you?

**Reply by kevinbluer** [reply]
hey @mrghubba! awesome re the feedback if / when you have any. also probably the easiest way to get in touch with technical question is here on Github discussions (which both myself and the team actively monitor). in addition @emilyJLin95 posted a great summary of how our support channels at the following too: https://trufflesuite.com/blog/truffle-community-updates-introducing-github-discussions

**Comment by mrghubba** [comment]
Okay. Sounds good
